src/components/auth-provider.tsx

```diff
@@ -35,7 +36,17 @@ const AuthProvider: React.FC< AuthProviderProps > = ( { children } ) => {
 	const [ user, setUser ] = useState< AuthContextType[ 'user' ] >( undefined );
 	const { locale } = useI18nData();
 
-	const authenticate = getIpcApi().authenticate;
+	const authenticate = useCallback( async () => {
+		try {
+			await getIpcApi().authenticate();
+		} catch ( error ) {
+			getIpcApi().showErrorMessageBox( {
+				title: __( 'Failed to open browser' ),
+				error,
+			} );
+		}
```

Comment:
I am thinking about creating HOF to avoid copy-paste, but in general it's an optional approach which I see.

Reply:
Calling `showErrorMessageBox` directly from `authenticate` and `openSiteURL` seems like a good idea 👍 They're both part of `ipc-handlers.ts`, so it should hopefully be relatively straightforward.